### PR TITLE
Stop multiple calls to AsyncPGInstrumentor.__init__ from clobbering instance attributes

### LINKED ISSUE #1790: Instantiating AsyncPGInstrumentor after .instrument has been called causes tracing to fail
**Describe your environment**

Python 3.11 on a M2 Mac.

**Steps to reproduce**

Run the following code (with `python -m asyncio` to allow top-level `async`/`await`):
```
import asyncpg

from opentelemetry.instrumentation.asyncpg import AsyncPGInstrumentor
from opentelemetry.sdk.trace import TracerProvider
from opentelemetry.sdk.trace.export import BatchSpanProcessor, ConsoleSpanExporter
from opentelemetry.trace import set_tracer_provider

provider = TracerProvider()
processor = BatchSpanProcessor(ConsoleSpanExporter())
provider.add_span_processor(processor)
set_tracer_provider(provider)

dsn = "postgres://postgres:password@localhost:54320/postgres"

AsyncPGInstrumentor().instrument()
AsyncPGInstrumentor()

connection = await asyncpg.connect(dsn)
await connection.execute("SELECT 1")
```

**What is the expected behavior?**

The SQL query runs successfully and a span is exported to the console.

**What is the actual behavior?**
What did you see instead?

```
AttributeError: 'NoneType' object has no attribute 'start_as_current_span'
```

**Additional context**

Each instantiation of `AsyncPGInstrumentor` runs `__init__`, which sets `self._tracer` to `None`. However, `BaseInstrumentor` overrides `__new__` to implement the singleton pattern, so only one instance of `AsyncPGInstrumentor` is ever created. Instantiating `AsyncPGInstrumentor` after `instrument` has been called (which sets `self._tracer`) therefore sets `self._tracer` back to `None`, which is a state inconsistent with `_is_instrumented_by_opentelemetry` (which will still be `True`).

A simple solution is to remove the line `self._tracer = None`.

### PR DESCRIPTION
**Pull request by marcuslimdw**
# Description

Don't set `_tracer` in `AsyncPGInstrumentor.__init__` is called to avoid corrupting singleton state and extract `leading_comment_remover` to class-level to prevent multiple compilations of a constant regex.

Fixes #1790

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

* Added a unit test (`test_duplicated_instrumentation_works`)
* Manually tested against a Postgres container (see code snippet in [issue](https://github.com/open-telemetry/opentelemetry-python-contrib/issues/1790))

# Does This PR Require a Core Repo Change?

- [ ] Yes. - Link to PR: 
- [x] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [x] Followed the style guidelines of this project
- [ ] Changelogs have been updated
- [x] Unit tests have been added
- [ ] Documentation has been updated

### CODE REVIEW

**Comment by andrijaperovic** [comment]
@marcuslimdw thank you for the fix! When is this expected to be merged?

**Comment by marcuslimdw** [comment]
@andrijaperovic not sure, it's waiting on review...

**Review by srikanthccv — APPROVED** [review]

**Review by shalevr — APPROVED** [review]

**Comment by shalevr** [comment]
Please fix the lint error and add an entry to the changelog

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   ([#2151](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2151))
 - `opentelemetry-resource-detector-azure` Added 10s timeout to VM Resource Detector
   ([#2119](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2119))
+- `opentelemetry-instrumentation-asyncpg` Allow AsyncPGInstrumentor to be instantiated multiple times
```
Comment (shalevr) [inline CHANGELOG.md:14]:
```suggestion
- `opentelemetry-instrumentation-asyncpg` Allow AsyncPGInstrumentor to be instantiated multiple times
([#1791](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1791))
```

**Pull request merged**